libs/datagrid/src/lib/cells/market-name-cell.tsx

```diff
@@ -1,39 +1,75 @@
-import type { MouseEvent, ReactNode } from 'react';
+import type { MouseEvent } from 'react';
 import { useCallback } from 'react';
 import get from 'lodash/get';
+import { Pill } from '@vegaprotocol/ui-toolkit';
+import type { Market } from '@vegaprotocol/types';
+
+const productTypeMap: { [key in string]: string } = {
+  Future: 'Futr',
+  Spot: 'Spot',
+  Perpetual: 'Perp',
+};
+
+export const MarketProductPill = ({
+  productType,
+}: {
+  productType?: string;
+}) => {
+  return productType ? (
+    <Pill size="xxs" className="uppercase ml-0.5" title={productType}>
+      {productTypeMap[productType] || productType}
+    </Pill>
+  ) : null;
+};
 
 interface MarketNameCellProps {
   value?: string | null;
-  data?: { id?: string; marketId?: string; market?: { id: string } };
+  data?:
+    | { id?: string; marketId?: string; productType?: string; market?: Market }
+    | Market;
   idPath?: string;
   onMarketClick?: (marketId: string, metaKey?: boolean) => void;
-  defaultValue?: ReactNode;
+  productType?: string;
 }
 
 export const MarketNameCell = ({
   value,
   data,
   idPath,
   onMarketClick,
+  productType,
 }: MarketNameCellProps) => {
   const id = data ? get(data, idPath ?? 'id', 'all') : '';
   const handleOnClick = useCallback(
     (ev: MouseEvent<HTMLButtonElement>) => {
       ev.preventDefault();
       ev.stopPropagation();
-      if (onMarketClick) {
-        onMarketClick(id, ev.metaKey || ev.ctrlKey);
-      }
+      id && onMarketClick?.(id, ev.metaKey || ev.ctrlKey);
     },
     [id, onMarketClick]
   );
-  if (!value || !data) return null;
-  return onMarketClick ? (
+
+  productType =
```

Comment:
(non blocking): The props of this component seem overly complex. Can we simplify in any way. I'd look into making this component take two values only, a product type string and a market name string

Reply:
I would prefer passing props like that, because it gives possibility to register the component in ag-grid and call renderer easily as string in the columns